controllers/src/main/java/com/redhat/ipaas/controllers/integration/StatusChangeHandlerProvider.java

```diff
@@ -36,33 +36,39 @@
         StatusUpdate execute(Integration model);
 
         class StatusUpdate {
-            private Optional<Integration.Status> status;
-            private Optional<String> statusMessage;
+            private Integration.Status status;
+            private String statusMessage;
+            private List<String> stepsPerformed;
 
-            public StatusUpdate(Optional<Integration.Status> status, Optional<String> statusMessage) {
+            public StatusUpdate(Integration.Status status, String statusMessage, List<String> stepsPerformed) {
                 this.status = status;
                 this.statusMessage = statusMessage;
-            }
-
-            public StatusUpdate(Optional<Integration.Status> status, String statusMessage) {
-                this(status, Optional.of(statusMessage));
+                this.stepsPerformed = stepsPerformed;
             }
 
             public StatusUpdate(Integration.Status status, String statusMessage) {
-                this(Optional.of(status), Optional.of(statusMessage));
+                this(status, statusMessage, null);
             }
 
             public StatusUpdate(Integration.Status status) {
-                this(Optional.of(status), Optional.empty());
+                this(status, null, null);
+            }
+
+            public StatusUpdate(Integration.Status status, List<String> stepsPerformed) {
+                this(status, null, stepsPerformed != null && stepsPerformed.size() > 0 ? stepsPerformed : null);
             }
 
-            public Optional<Integration.Status> getStatus() {
+            public Integration.Status getStatus() {
                 return status;
             }
 
-            public Optional<String> getStatusMessage() {
+            public String getStatusMessage() {
```

Comment:
This should be an optional, otherwise your get NPES.
```
java.lang.NullPointerException: statusMessage
	at java.util.Objects.requireNonNull(Objects.java:228) ~[na:1.8.0_111]
	at com.redhat.ipaas.model.integration.ImmutableIntegration$Builder.statusMessage(ImmutableIntegration.java:2459) ~[model-0.1-SNAPSHOT.jar!/:na]
	at com.redhat.ipaas.controllers.integration.IntegrationController.lambda$callStatusChangeHandler$10(IntegrationController.java:179) ~[controllers-0.1-SNAPSHOT.jar!/:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_111]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_111]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_111]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_111]


```